crates/compiler/mono/src/drop_specialization.rs

```diff
@@ -848,20 +1020,16 @@ fn get_union_tag_layout(union_layout: UnionLayout<'_>, tag: Option<Tag>) -> Unio
 Branch on the uniqueness of a symbol.
 Using a joinpoint with the continuation as the body.
 */
-fn branch_uniqueness<'a, 'i, F1, F2>(
+fn branch_uniqueness<'a, 'i>(
     arena: &'a Bump,
     ident_ids: &'i mut IdentIds,
     layout_interner: &'i mut STLayoutInterner<'a>,
     environment: &DropSpecializationEnvironment<'a>,
     symbol: Symbol,
-    unique: F1,
-    not_unique: F2,
+    unique: impl FnOnce(&mut STLayoutInterner<'a>, &mut IdentIds, &'a Stmt<'a>) -> &'a Stmt<'a>,
+    not_unique: impl FnOnce(&mut STLayoutInterner<'a>, &mut IdentIds, &'a Stmt<'a>) -> &'a Stmt<'a>,
     continutation: &'a Stmt<'a>,
-) -> &'a Stmt<'a>
-where
-    F1: FnOnce(&mut STLayoutInterner<'a>, &mut IdentIds, &'a Stmt<'a>) -> &'a Stmt<'a>,
-    F2: FnOnce(&mut STLayoutInterner<'a>, &mut IdentIds, &'a Stmt<'a>) -> &'a Stmt<'a>,
-{
+) -> &'a Stmt<'a> {
```

Comment:
is this really a win in terms of readability?

Reply:
Perhaps not, but consistency wise maybe. I didn't know this was an option, otherwise I would have done it this way the first time. But i can change it back.

Reply:
well for simple things like `impl Write`  or `impl Eq` I think it's a clear win. but with types this complex, the where is cleaner to me personally. I also like how things line up. but totally your call, this is a matter of personal taste